source/common/network/io_socket_handle_impl.cc

```diff
@@ -477,6 +481,15 @@ Api::SysCallIntResult IoSocketHandleImpl::setBlocking(bool blocking) {
   return Api::OsSysCallsSingleton::get().setsocketblocking(fd_, blocking);
 }
 
+IoHandlePtr IoSocketHandleImpl::duplicate() {
+  auto result = Api::OsSysCallsSingleton::get().duplicate(fd_);
+  if (result.rc_ == -1) {
+    throw EnvoyException(fmt::format("duplicate failed for '{}': ({}) {}", fd_, result.errno_,
```

Comment:
It could fail if the process bumps into fd limit. We have RELEASE_ASSERT for failing `socket` calls, so the same approach should be taken here as well.

Reply:
I made it into a RELEASE_ASSERT here instead at 13ff61610c48391cf70ae52dbc51b2d88c2132dd